src/Adapter/Product/Repository/ProductMultiShopRepository.php

```diff
@@ -169,6 +170,10 @@ public function getProductDefaultShopId(ProductId $productId): ShopId
      */
     public function getByShopConstraint(ProductId $productId, ShopConstraint $shopConstraint): Product
     {
+        if ($shopConstraint->getShopGroupId()) {
+            throw new InvalidShopConstraintException('Product has no features related with shop group use single shop and all shops constraints');
```

Comment:
> use single shop and all shops constraints

Did you mean: use single shop OR all shops constraints?

Reply:
Indeed or would be more adapted, do you want me to fix it?

Reply:
no need 👍